lib/Driver/Doctrine/DBAL/DBALAbstractField.php

```diff
@@ -0,0 +1,141 @@
+ */
+
+namespace FSi\Component\DataSource\Driver\Doctrine\DBAL;
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use FSi\Component\DataSource\Driver\Doctrine\DBAL\Exception\DBALDriverException;
+use FSi\Component\DataSource\Field\FieldAbstractType;
+
+abstract class DBALAbstractField extends FieldAbstractType implements DBALFieldInterface
+{
+    public function buildQuery(QueryBuilder $qb, $alias)
+    {
+        $data = $this->getCleanParameter();
+        $fieldName = $this->getFieldName($alias);
+        $name = $this->getName();
+
+        if (($data === array()) || ($data === '') || ($data === null)) {
+            return;
+        }
+
+        $type = $this->getDBALType();
+        $comparison = $this->getComparison();
+
+        $clause = $this->getOption('clause');
+        $func = sprintf('and%s', ucfirst($clause));
+
+        if ($comparison == 'between') {
+            if (!is_array($data)) {
+                throw new DBALDriverException('Fields with \'between\' comparison require to bind an array.');
+            }
+
+            $from = array_shift($data);
+            $to = array_shift($data);
+
+            if (empty($from) && ($from !== 0)) {
+                $from = null;
+            }
+
+            if (empty($to) && ($to !== 0)) {
+                $to = null;
+            }
+
+            if ($from === null && $to === null) {
+                return;
+            } elseif ($from === null) {
+                $comparison = 'lte';
+                $data = $to;
+            } elseif ($to === null) {
+                $comparison = 'gte';
+                $data = $from;
+            } else {
+                $qb->$func("$fieldName BETWEEN :{$name}_from AND :{$name}_to");
+                $qb->setParameter("{$name}_from", $from, $type);
+                $qb->setParameter("{$name}_to", $to, $type);
+                return;
+            }
+        }
+
+        if ($comparison == 'isNull') {
+            $qb->$func($fieldName . ' IS ' . ($data === 'null' ? '' : 'NOT ') . 'NULL');
+            return;
+        }
+
+        $expr = $qb->expr();
+        if (in_array($comparison, array('in', 'notIn'))) {
+            if (!is_array($data)) {
+                throw new DBALDriverException('Fields with \'in\' and \'notIn\' comparisons require to bind an array.');
+            }
+            $placeholders = array();
+            foreach ($data as $value) {
+                $placeholders[] = $qb->createNamedParameter($value);
+            }
+            //this is because "in" and "notIn" was added in DBAL 2.4
+            $comparison = $comparison === 'in' ? 'IN' : 'NOT IN';
+            $qb->$func($expr->comparison($fieldName, $comparison, '('.implode(', ', $placeholders).')'));
+            return;
+        } elseif (in_array($comparison, array('like', 'contains'))) {
+            $data = "%$data%";
+            $comparison = 'like';
+        }
+
+        $qb->$func($expr->$comparison($fieldName, ":$name"));
```

Comment:
Wouldn't hurt to check if the `$func` is actually callable on the query builder.